pkg/middlewares/aws_defaults.go

```diff
@@ -0,0 +1,204 @@
+package middlewares
+
+import (
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+const defaultIamRolePathGlob = "/aws-service-role/*"
+
+// When scanning a brand new AWS account, some users may see irrelevant results about default AWS role policies.
+// We ignore these resources by default when strict mode is disabled.
+type AwsDefaults struct{}
+
+func NewAwsDefaults() AwsDefaults {
+	return AwsDefaults{}
+}
+
+func awsIamRoleDefaults(remoteResources, resourcesFromState *[]resource.Resource) error {
+	newRemoteResources := make([]resource.Resource, 0)
+
+	for _, remoteResource := range *remoteResources {
+		// Ignore all resources other than iam role
+		if remoteResource.TerraformType() != aws.AwsIamRoleResourceType {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		existInState := false
+		for _, stateResource := range *resourcesFromState {
+			if resource.IsSameResource(remoteResource, stateResource) {
+				existInState = true
+				break
+			}
+		}
+
+		if existInState {
+			newRemoteResources = append(newRemoteResources, remoteResource)
+			continue
+		}
+
+		match, err := filepath.Match(defaultIamRolePathGlob, *remoteResource.(*aws.AwsIamRole).Path)
```

Comment:
Why not doing a simple `strings.HasPrefix(*remoteResource.(*aws.AwsIamRole).Path, "/aws-service-role/")` I don't think we need a wildcard match there